docs/xenit-style-guide/golang.md

```diff
@@ -132,52 +136,130 @@ case args.Push != nil:
-        panic("An unexpected error happen!")
-    })
+The logging object should be passed through the context when calling a function that needs to log. This removes the need to add an additional parameter that needs to be tracked.
+
+```golang
+ctx := logr.NewContext(context.Background(), log)
+run(ctx)
 
-    // Listen and Server in 0.0.0.0:8080
-    r.Run(":8080")
+func run(ctx context.Context) {
+	log := logr.FromContextOrDiscard(ctx)
+	log.Info("running function")
 }
 ```
 
-### Datadog
+## HTTP
 
-Using the `status` field when logging can interfere with how Datadog interprets the severity of the log. If you have a log with fields `status = 400` and `level = error`, then Datadog reports this as an info log. If you use another field for the status code, for example `http_response.status_code`, together with `level = error`, Datadog will report it as an error log. You can read more about the `status` field and other reserved log fields in Datadog [here](https://docs.datadoghq.com/logs/log_configuration/attributes_naming_convention/#reserved-attributes).
+The Go standard library includes a http package that works well for simple applications, but requires a lot of custom code when building larger projects. While it is fine to just use the standard library for simple applications it is preferable to switch to [Gin](https://github.com/gin-gonic/gin) as project feature requirements develop. Gin provides extra functionality and extensions to simplify things like parsing parameters in the URL path and endpoint authorization.
+
+The HTTP server should be started and stopped in accordance to the Startup and Shutdown documentation. The HTTP server should be gracefully stopped before the program exits to make sure that all in flight requests are processed. A good practice is to use the [logger middleware](https://github.com/XenitAB/pkg/tree/main/middleware/logger) to log all incoming requests to the HTTP server.
+
+```golang
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/alexflint/go-arg"
+	"github.com/gin-gonic/gin"
+	"github.com/go-logr/zapr"
+	"github.com/xenitab/pkg/middleware/logger"
+	"go.uber.org/zap"
+)
+
+type arguments struct {
+	Addr string `arg:"--addr" default:":8080"`
+}
+
+func main() {
+	zapLog, err := zap.NewProduction()
+	if err != nil {
+		panic(fmt.Sprintf("who watches the watchmen (%v)?", err))
+	}
+	log := zapr.NewLogger(zapLog)
+
+	args := &arguments{}
+	arg.MustParse(args)
+
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer cancel()
+	g, ctx := errgroup.WithContext(ctx)
+
+	gin.SetMode(gin.ReleaseMode)
+	router := gin.New()
+	router.Use(logger.Logger(log))
+	router.Use(gin.Recovery())
+	router.GET("/ping", func(c *gin.Context) {
+		c.JSON(http.StatusOK, gin.H{
+			"message": "pong",
+		})
+	})
+	srv := &http.Server{
+		Addr:    args.Addr,
+		Handler: router,
+	}
+	g.Go(func() error {
```

Comment:
Do you want to explain why you want to use error groups?

I.e. why can't I just close these serially and make the cose slightly easier to understand?